### PR TITLE
[[ Bug 13658 ]] Data corrupted by the shell() function on server

### PR DESCRIPTION
**Pull request by livecodesebastien**

### CODE REVIEW

**Comment by livecodesebastien** [comment]
This should probably wait for 6.7-rc-4 (or 6.7-GM-1) built, since the bug is not fixed in develop

File: engine/src/exec-files.cpp [hunk 1]
```diff
@@ -2118,6 +2118,14 @@ void MCFilesExecReadFromProcess(MCExecContext& ctxt, MCNameRef p_process, MCStri
     t_encoding = MCprocesses[t_index].encoding;
 	MCAutoValueRef t_output;
 
+    // SN-2014-10-14: [[ Bug 13658 ]] In case we want to read everything (EOF, end, empty) from the process,
+    //  the sentinel must be empty, not Ctrl-D (0x04, which might appear in a binary data output.
+    MCAutoStringRef t_sentinel;
+    if (MCStringGetLength(p_sentinel) == 1 && MCStringGetCharAtIndex(p_sentinel, 0) == 0x4)
```
Comment (runrevmark) [inline engine/src/exec-files.cpp]: I think we need to keep the existing behavior here in 'text mode' - in that mode either real EOF or numToChar(4) should cause eof.

File: engine/src/sysspec.cpp [hunk 2]
```diff
@@ -1643,11 +1643,16 @@ IO_stat MCS_runcmd(MCStringRef p_command, MCStringRef& r_output)
     MCAutoStringRef t_data_string;
     // MW-2013-08-07: [[ Bug 11089 ]] The MCSystem::Shell() call returns binary data,
 	//   so since uses of MCS_runcmd() expect text, we need to do EOL conversion.
-    if (!MCStringCreateWithNativeChars((char_t*)MCDataGetBytePtr(*t_data), MCDataGetLength(*t_data), &t_data_string) ||
-        (!MCStringConvertLineEndingsToLiveCode(*t_data_string, r_output)))
+    // SN-2014-10-14: [[ Bug 13658 ]] Get the behaviour back to what it was - no line-ending conversion for desktop
```
Comment (runrevmark) [inline engine/src/sysspec.cpp]: I don't think this is correct - we fixed EOL conversion for shell() in 6.1.1 on Desktop... i.e. shell() currently is defined as returning 'text', therefore it must be EOL conversion in the same way as the other file functions on all platforms / modes.

**Comment by runrevmark** [comment]
Hi Seb - okay I checked back... We need to do line ending conversion in shell() on Windows and on Server - i.e. not Mac/Linux desktop. The rationale here is that UNIX line endings (which is typically what most apps you shell() to use) are numToChar(10) which matches normalized line endings. Thus windows is the only case where we wouldn't get normalized line endings.

**Pull request merged**